rflx/pyrflx/typevalue.py

```diff
@@ -228,10 +228,10 @@ def assign(self, value: str, check: bool = True) -> None:
         if Variable(prefixed_value) not in self.literals:
             raise KeyError(f"{value} is not a valid enum value")
         r = (
-            And(*self._type.constraints("__VALUE__", check))
+            And(*self._type.constraints("__VALUE__", check, not self.__imported))
             .substituted(
                 mapping={
-                    **{Variable(k): v for k, v in self._type.literals.items()},
+                    **self.literals,
```

Comment:
I think we should use `self.__literals` here since it is used everywhere else inside the class.

Reply:
That is not true. `self.literals` is used four times in this class - it is always used when the attribute is just accessed. `self.__literals` is only used for its initialization and for the property definition.

I generally (attempt to) use a property for accessing an attribute, as it gives a bit more protection: Mutable attributes (e.g. a list) cannot be changed by a called function. This has of course a performance impact, but I suppose it is rather small.